### PR TITLE
Automate ec2 plugin

### PR DESCRIPTION
**Pull request by michaelyoungstrom**
This script automates the ec2 plugin setup based on the `ec2_config.yml` file, which can include 1 or multiple ec2 clouds and AMIs

### CODE REVIEW

File: src/main/groovy/4configureEc2Plugin.groovy [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+String configPath = System.getenv("JENKINS_CONFIG_PATH")
+try {
+    configText = new File("${configPath}/ec2_config.yml").text
+} catch (FileNotFoundException e) {
+    logger.severe("Cannot find config file path @ ${configPath}/ec2_config.yml")
+    jenkins.doSafeExit(null)
+    System.exit(1)
+}
+ec2Config = yaml.load(configText)
+ec2CloudConfig = ec2Config.CLOUDS
```
Comment (estute) [inline src/main/groovy/4configureEc2Plugin.groovy]: I think we should type as often as possible. It makes scripts clearer, and adds another layer of protection, whereas the inference typing in groovy can obfuscate issues with the data we pass in

File: src/main/groovy/4configureEc2Plugin.groovy [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+List<AmazonEC2Cloud> clouds = new ArrayList<AmazonEC2Cloud>();
+for (cloudConfig in ec2CloudConfig) {
+    List<SlaveTemplate> templates = new ArrayList<SlaveTemplate>();
+    ec2AmiConfig = cloudConfig.AMIS
+    // Loop through the AMIs
+    for (amiConfig in ec2AmiConfig) {
+        // Create Spot Config if applicable
+        maxBid = amiConfig.SPOT_CONFIG.SPOT_MAX_BID_PRICE
+        // The constructor changes based on the plugin so add support for both
+        if (pluginVersion < 1.32) {
```
Comment (estute) [inline src/main/groovy/4configureEc2Plugin.groovy]: Unfortunately, I suspect we will have to do this type of thing more often, as we upgrade plugins

File: src/main/groovy/4configureEc2Plugin.groovy [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            bidType = amiConfig.SPOT_CONFIG.SPOT_INSTANCE_BID_TYPE.toLowerCase()
+            validBidTypes = ['persistent', 'one-time']
+            if (maxBid && bidType) {
+                if (!(bidType in validBidTypes)) {
+                    logger.severe("Invalid value for SPOT_INSTANCE_BID_TYPE. Must be " +
+                                  "persistent, one-time, or null if no spot configuration " +
+                                  "is desired. Got: ${bidType}")
+                    //Create SpotConfiguration
```
Comment (estute) [inline src/main/groovy/4configureEc2Plugin.groovy]: Shouldn't we be aborting init if we have invalid values passed in?

Comment (michaelyoungstrom) [inline src/main/groovy/4configureEc2Plugin.groovy]: Good point. Accidentally changed this during testing.

**Pull request merged**